src/RemoteInboxNotifications/GetRuleProcessor.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Gets the processor for the specified rule type.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\DateTimeProvider\CurrentDateTimeProvider;
+use \Automattic\WooCommerce\Admin\PluginsProvider\PluginsProvider;
+
+/**
+ * Class encapsulating getting the processor for a given rule type.
+ */
+class GetRuleProcessor {
+	/**
+	 * Get the processor for the specified rule type.
+	 *
+	 * @param string $rule_type The rule type.
+	 *
+	 * @return object The matching processor for the specified rule type, or a FailRuleProcessor if no matching processor is found.
+	 */
+	public static function get_processor( $rule_type ) {
+		switch ( $rule_type ) {
+			case 'plugins_activated':
+				return new PluginsActivatedRuleProcessor(
+					new PluginsProvider()
```

Comment:
Assuming this injection technique is just to ease testing, what do you think about these becoming defaults in the rule processor constructors rather than having to pass them in here?

Reply:
Done in d1fffcae